Update library to reflect teleport install systemd addition

sagesyr opened the pull request:
- Updates the [Run Teleport as a Daemon](https://goteleport.com/docs/setup/admin/daemon/) guide to reflect changes to the process of setting up a daemon when using the new `telelport install systemd` command
- Adds new daemon install steps to our [Enterprise Getting Started Guide](https://goteleport.com/docs/enterprise/getting-started/)
- Adds new daemon install steps  to our [Getting Started on a Linux Server Guide](https://goteleport.com/docs/getting-started/linux-server/)

russjones committed:
Removed gocheck from fixtures

Removed gocheck from fixtures as gocheck is no longer used.

russjones committed:
Refactor tests under bpf tests

Refactored all tests under "lib/bpf" and "lib/restrictedsession" to use
testify instead of gocheck.

fspmarshall committed:
cluster alerts

russjones committed:
Fix.

Commit:
Update library to reflect teleport install systemd addition

Commit:
Update library to reflect teleport install systemd addition

Commit:
correct steps in getting started

Commit:
correct steps in getting started

zmb3 commented on docs/pages/setup/admin/daemon.mdx:
```diff
@@ -31,117 +35,174 @@ $ readlink /sbin/init
 
-Copy the recommended Teleport unit file for `systemd` below and paste the
-contents into a file called `/usr/lib/systemd/system/teleport.service`:
+The contents of the default unit file will reflect the following defaults:
```
Do you think we need to put the contents of the file here?

One of the reasons for this new command is to reduce the number of duplicated systemd examples we have throughout the code and the docs. If we list the contents here, we have to keep it in sync with future changes.

Maybe instead we could encourage the user to run `teleport install systemd` without the `-o` flag if they want to inspect the default unit file.

rsyracuse replied:
This is a really good point in keeping things evergreen, I don't see why the contents of the unit file would be necessary in this document and removed it.

zmb3 commented on docs/pages/setup/admin/daemon.mdx:
````diff
@@ -31,117 +35,174 @@ $ readlink /sbin/init
 ```
 
-<Notice type="tip">
+Users who rely on DEB or RPM can enter the following commands to enable and start the systemd `teleport` daemon at any time. Enabling the daemon using `systemctl` allows systemd to place the daemon in it's dependancy tree, ensuring that the teleport service will be automatically restarted on any potential reboot:
````
```suggestion
Users who rely on DEB or RPM can enter the following commands to enable and start the systemd `teleport` daemon at any time. Enabling the daemon using `systemctl` allows systemd to place the daemon in its dependancy tree, ensuring that the teleport service will be automatically restarted on any potential reboot:
```

zmb3 commented on docs/pages/setup/admin/daemon.mdx:
````diff
@@ -31,117 +35,174 @@ $ readlink /sbin/init
 
 ```code
-$ sudo systemctl start teleport.service
+[Unit]
````
Same comment as above - I'd prefer to not have this snippet pasted in a bunch of different places.

rsyracuse replied:
This has been removed

klizhentas commented on docs/pages/getting-started/linux-server.mdx:
```diff
@@ -166,12 +166,26 @@ On your Linux machine, run the following command to start the `teleport` daemon
 
   <TabItem label="Source or custom install">
+  1. Create a systemd unit file using the `teleport install systemd` command and the [unit file path](https://www.freedesktop.org/software/systemd/man/systemd.unit.html) you'd
+  like to use for your configuration. The `/etc/systemd/system/teleport.service` path is recommended in most use case:
```
```suggestion
  like to use for your configuration. We recommend `/etc/systemd/system/teleport.service` path in most use cases:
```

klizhentas replied:
style nit: fixed the typo and switched to active voice.

rsyracuse replied:
Switched to the active voice here and in more places, fixed typo

klizhentas commented on docs/pages/getting-started/linux-server.mdx:
```diff
@@ -166,12 +166,26 @@ On your Linux machine, run the following command to start the `teleport` daemon
 
+<Admonition title="The Teleport daemon" type="tip">
+Configuration for the teleport daemon relies on [systemd](https://www.freedesktop.org/wiki/Software/systemd/). For more information on how the
+teleport service daemon is configured, see our guide on how to [Run Teleport as a Daemon](https://goteleport.com/docs/setup/admin/daemon/).
```
```suggestion
teleport service daemon is configured, see our guide on how to [Run Teleport as a Daemon](../setup/admin/daemon.mdx).
```

klizhentas replied:
Using internal links is better when possible, we will verify them more efficiently.

rsyracuse replied:
Good point, I completed this for all `goteleport.com` links

klizhentas commented on docs/pages/getting-started/linux-server.mdx:
```diff
@@ -166,12 +166,26 @@ On your Linux machine, run the following command to start the `teleport` daemon
   </TabItem>
 
   <TabItem label="Source or custom install">
+  1. Create a systemd unit file using the `teleport install systemd` command and the [unit file path](https://www.freedesktop.org/software/systemd/man/systemd.unit.html) you'd
```
I recommend moving additional articles and links to the last section "Next steps" to avoid [breaking readers focus](https://goteleport.com/docs/contributing/documentation/style-guide/#expectations) according to our style guide.

rsyracuse replied:
Moved all links to the "Next Steps" section, retitled from "Further Reading" to better fit our style guide

klizhentas commented on docs/pages/setup/admin/daemon.mdx:
````diff
@@ -31,117 +35,174 @@ $ readlink /sbin/init
 ```
 
-## Step 2/3. Create a systemd unit file
+## Creating and Configuring the systemd Daemon
````
@ptgott @rsyracuse let's keep the Step 2/3 structure for now. I plan to add a special plugin that will render this differently as well similar to Vault's docs to communicate that it's only 3 steps total and you are almost there (2/3)

<img width="245" alt="Screen Shot 2022-08-17 at 4 15 43 PM" src="https://user-images.githubusercontent.com/13717/185259787-20e8eb2f-1809-422a-99fa-eec9d5792f12.png">

rsyracuse replied:
Re-added the steps, this makes a lot of sense

klizhentas commented on docs/pages/setup/admin/daemon.mdx:
```diff
@@ -31,117 +35,174 @@ $ readlink /sbin/init
-## Step 2/3. Create a systemd unit file
+## Creating and Configuring the systemd Daemon
+
+Once you've initiially installed the `teleport` binary and created a Teleport configuration file, the Teleport daemon must be created and configured. While a daemon can be created and configured manually, the
```
```suggestion
Once you've initially installed the `teleport` binary and created a Teleport configuration file,  let's create and configure Teleport daemon.
```

klizhentas replied:
nit: switched from passive to active and shortened the sentence according to [On writing well](https://www.goodreads.com/book/show/53343.On_Writing_Well)

rsyracuse replied:
Searched for other opportunities to switch to the active voice, including your request and some others

klizhentas commented on docs/pages/setup/admin/daemon.mdx:
```diff
@@ -31,117 +35,174 @@ $ readlink /sbin/init
+
+### Creating the systemd Daemon following Package Manager Installation
+
+If Teleport is being installed using a DEB or RPM package manager, a `teleport` systemd daemon configuration will have been included as part of the teleport package installation by default. To check that this daemon was installed correctly,
```
```suggestion
We included a default `teleport` daemon configuration in DEB or RPM package manager. To check that this daemon was installed correctly,
```

rsyracuse replied:
Abbreviated this section largely as requested, this has much better flow

klizhentas reviewed (commented):
@rsyracuse thank you for the PR, I think it's almost ready to merge, but I left a couple of nitpicks and suggestions

russjones committed:
Fix.

Commit:
Add Windows license negotation to Desktop Access

Commit:
enterprise install steps/recommended fixes

Commit:
enterprise install steps/recommended fixes

Commit:
next steps section fix

Commit:
next steps section fix

Commit:
typo fix

Commit:
typo fix

Commit:
Merge branch 'master' into 15394-teleport-install-systemdcli

Commit:
link fix

Commit:
link fix

Commit:
fixlink

Commit:
fixlink

Commit:
Merge branch 'master' into 15394-teleport-install-systemdcli

tobiaszheller committed:
Delete touch_id credentials during tsh mfa rm (#15459)

Delete touch_id credentials of tsh mfa rm

Commit:
Add an AWS EC2 instance fetcher (#13886)

* Add an AWS EC2 instance fetcher

* Resolve comments

- Remove leftover VPCID stuff
- Document public functions
- Return a notfound when there are no nodes fetched instead of nil nil

* Resolve comments

* Fix fileconf_test.go

logand22 committed:
Replace quay.io with amazon ECR where appropriate (#15382)

rosstimothy committed:
Fix race in EC2 label warning (#15668)

EC2 loggedInstanceTagsNotFound was unprotected and susceptible to
a race from Start and Sync. To mitigate this we can use a sync.Once
instead of maintaining a boolean flag.

stevenGravy committed:
Misspelling fixes (#15672)

michellescripts committed:
[auto] Update webassets in teleport/master from webassets/master (#15671)

ptgott committed:
Fix table formatting in the SOC 2 guide (#15660)

Some rows of the table within the guide include separate table columns
for different Teleport features that support a single SOC 2 requirement.
The table applies this format inconsistently, causing some table cells
to be hidden.

This change limits each table row to three columns--the number of
columns defined in the first row of the table--and uses line breaks to
separate multiple features that support a single SOC 2 requirement.

Commit:
bash formatting

Commit:
bash formatting

Commit:
Merge branch '15394-teleport-install-systemdcli' of https://github.com/rsyracuse/teleport into 15394-teleport-install-systemdcli

GavinFrazar committed:
Azure mysql postgres auto discovery configuration (#15629)

* Add Azure auto-discovery configuration fields

* Init databases if azure matchers are in config

* Use AzureMatchers in db service

* Use all azure subscriptions/resource groups if omitted in matcher

* Add azure config tests

* Update lib/services/matchers.go

Co-authored-by: Krzysztof Skrzętnicki <krzysztof.skrzetnicki@goteleport.com>

* Update lib/config/fileconf.go

Co-authored-by: Marek Smoliński <marek@goteleport.com>

* Update lib/config/fileconf.go

Co-authored-by: Marek Smoliński <marek@goteleport.com>

* Update lib/services/matchers.go

Co-authored-by: Marek Smoliński <marek@goteleport.com>

* Remove superfluous cmp option for diffing azure matcher

* Rename AzureMatchers Tags to ResourceTags

* Deduplicate subscription/resource groups and add tests

* Remove azure matcher config fixup

Co-authored-by: Krzysztof Skrzętnicki <krzysztof.skrzetnicki@goteleport.com>
Co-authored-by: Marek Smoliński <marek@goteleport.com>

GavinFrazar committed:
Azure mysql postgres auto discovery config create (#15630)

* Add Azure auto-discovery configuration fields

* Init databases if azure matchers are in config

* Use AzureMatchers in db service

* Use all azure subscriptions/resource groups if omitted in matcher

* Add azure config tests

* Add config create flags for azure matchers

* Add config create tests for azure

* Move discovery flags for azure below aws

* Fixup merge

ptgott commented on docs/pages/setup/admin/daemon.mdx:
```diff
@@ -143,15 +163,44 @@ To upgrade a host to a newer version of Teleport, you must:
 
-## Further reading
+
+## Understanding Teleport Daemon Command Line Options for Unit File Configuration
```
Should this content be in the [CLI reference](https://github.com/gravitational/teleport/blob/d6e3202a49627d3e3043c5ebd7545e4fc5b64da3/docs/pages/setup/reference/cli.mdx)? Then we can add a link to this section of the CLI reference in the "Next steps" section.

rsyracuse replied:
It should be, but for now the CLI reference, specifically for the `teleport` command appears to be out of date beyond just the daemon config, and I felt it may cause friction if we link to a patchwork fix for the CLI reference to address systemd without a more focused effort for `teleport` subcommands as a whole. I created a separate issue to address this and it's next on my list of things to do should no other task need to be prioritized first. 

For the time being I felt it would be best to update the doc highlighting the feature directly until the CLI reference gets a larger overhaul since other features of the CLI users may need aren't documented, but I am happy to begin that work now within this PR if you feel that may be better. 

https://github.com/gravitational/teleport/issues/15592

ptgott commented on docs/pages/setup/admin/daemon.mdx:
````diff
@@ -143,15 +163,44 @@ To upgrade a host to a newer version of Teleport, you must:
+--output=/etc/systemd/system/teleport.service
+```
+
+## Next Steps
````
```suggestion
## Next steps
```
Since we use sentence case for headings

ptgott commented on docs/pages/setup/admin/daemon.mdx:
```diff
@@ -143,15 +163,44 @@ To upgrade a host to a newer version of Teleport, you must:
+[Configuration Reference](../reference/config.mdx) to create your configuration file.
+
+If you need information on how `systemctl reload teleport` works, see our guide to
+[upgrading a teleport binary](./upgrading-the-teleport-binary.mdx).
```
```suggestion
[upgrading a Teleport binary](./upgrading-the-teleport-binary.mdx).
```

Commit:
unit test fix

Commit:
unit test fix

reedloden committed:
ci: Reduce CodeQL max goroutines to address failed extraction (#15683)

Ref https://github.com/github/codeql/issues/9888

CodeQL was failing intermittently due to exceeding limits when
going through the Go extraction phase. Reduce the concurrent
extractions by half (32 --> 16) in an attempt to fix this.

reedloden committed:
ci: Add paths/paths-ignore to GitHub Actions workflows to reduce unnecessary builds (#15636)

We're running CIFuzz and CodeQL in many cases where it doesn't make sense. We get
charged for GitHub Actions minutes used, so let's make sure to only run them when
absolutely necessary.

For CIFuzz, only run if a *.go file has been modified.
For CodeQL, only run if a path outside of docs/* and rfd/* has been modified.

Commit:
recommended fixes

Commit:
recommended fixes

Commit:
fix systemd ent description

Commit:
fix systemd ent description

rsyracuse commented on docs/pages/setup/admin/daemon.mdx:
```diff
@@ -33,89 +36,106 @@ $ readlink /sbin/init
+
+If the output describes a good configuration for your environment, enter the command once again using the `-o` flag to
+write the output to the directory path of the unit file itself.
+The file path `/etc/systemd/system/teleport.service` is recommended in most use cases as the optimal **unit file load path**,
```
I played around with the wording of this, but had a hard time finding a way to mention the recommended load path following the command itself that didn't feel as though it could be confusing to readers. I generally try to describe commands fully before guiding readers to enter them both so that readers can always intuit when the guide is moving on to the next step, and to avoid situations where readers may enter a command that changes their configuration before having an understanding of what the command will be doing. The risk of using the wrong load path in this guide is pretty low, but I do still feel it's worth mentioning for fringe cases

stevenGravy committed:
Added sles as another identifier for suse in auto install (#15659)

Added sles as another identifies for suse installation to support Suse 12 installs

codingllama committed:
Correctly handle U2F public key typecast failure (#15689)

U2F keys are always [ECDSA/P-256 keys][1], making this an impossible branch in
practice, but better safe than sorry.

[1]: https://fidoalliance.org/specs/fido-u2f-v1.2-ps-20170411/fido-u2f-raw-message-formats-v1.2-ps-20170411.html#registration-response-message-success

espadolini committed:
Run `go mod tidy` (#15711)

logand22 committed:
Update teleport plugin docs for Amazon ECR Public (#15358)

Plugin ECR Documenation updates

greedy52 committed:
Show proper error message when "tsh db env/config" are not supported (#14971)

ryanclark committed:
Record when a session recording is accessed (#14302)

This adds a new SessionRecordingAccess event that is emitted every time a session's events are queried, or if a session's events are streamed.

This event is emitted by both v1 and v2 of the API, so both the web UI and tsh play will result in an event in the audit log.

Implements #13880.

The frontend change for supporting this event type is here - gravitational/webapps#970.

Commit:
Add support for IdP-Initiated SAML2 login (#13924)

Commit:
Remove unused vars in `buffer.go` (#15553)

ptgott committed:
Organize docs guide sections chronologically (#15357)

* Organize docs guide sections chronologically

This change aims to make docs navigation easier by organizing some docs
sections according to the sequence of steps a user would take to set up
Teleport.

The current docs organization uses a variety of categories and schemes
to organize the docs. For example, there is a "Home" section that
includes the Changelog, Installation page, and Getting Started guides; a
"Setup" section that includes references and admin guides; and
edition-specific sections (Enterprise, Cloud). For a user who is setting
up Teleport--or who has already done some setup work and wants more
advanced instructions--it's difficult to know where in the docs to find
the right information.

This change organizes our how-to guides into the following categories
that describe the process of setting up Teleport:

- Try out Teleport
- Deploy a Cluster (including choosing an edition)
- Configure Access (including SSO, RBAC, and Access Requests)
- Manage your Cluster (admin guides, operations, etc.)
- Use Teleport (this section already exists)

I moved the Reference section after this chronology, since users can
access the reference guides anywhere in the setup process.

As part of the change, I have also moved the content from the
"Enterprise" and "Cloud" sections into "Deploy a Cluster", since this
content has to do with how to deploy a specific edition of Teleport.

Note that this change does _not_ attempt to reorganize our
protocol-specific sections. While adding resources is part of the
Teleport setup process, we have a lot of content in our
protocol-specific sections, and moving it all into a single section
related to adding resources to a cluster would (a) exceed the maximum
depth for subsections in the nav bar and (b) cause more confusion than
it alleviates.

* Respond to PR feedback

- Create a "Compliance Frameworks" section of "Configure Access" with
  the FedRAMP and SOC 2 guides
- Rename "Use Teleport" to "Connect your Client"
- Move the database GUI client guide into "Connect your Client"

* Add redirects

* Fix linter issues

r0mant committed:
(buddy) Pass JWT headers on websocket requests (#15667)

* transport: Rewrite headers, including JWTs, for websockets.

Applications can otherwise 401 on websocket requests, as they do not
present any authentication headers.

docs: Fix the reserved JWT header name.

Signed-off-by: Roman Tkachenko <roman@goteleport.com>

* Add test for JWT header in websocket apps

Signed-off-by: Roman Tkachenko <roman@goteleport.com>
Co-authored-by: Alex Vandiver <alex@chmrr.net>

tcsc committed:
Adds CI hooks for GLibc compatibility check (#15547)

Adds the GCB build yaml for controlling the build, and updates the test script
to work in both the GCB environment and on a local dev machine.

Also changes the centos buildbox to leave the default user as root. When
GCB mounts the workspace into the container, the source code is owned
by root, and there is no way to change this. This means that the build will
fail when the non-root user specified in the build image attempt to write files
into the workspace. Setting the root user fixes this.

See-Also: #15186

marcoandredinis committed:
NodeJoin script: fix when no labels are provided (#15709)

Recently we added a way to add labels on newly added nodes based on the
token.
https://github.com/gravitational/teleport/pull/15114
Each token now has a list of SuggestedLabels, which are used to feed
that list.

However, if that list is empty, the generated script would trigger the
following error when running the `teleport node configure ...` command: 
`teleport: error: unexpected`

This happens because the command is generating an empty argument `""`
when running the `teleport node configure ...` command.
So it looks like this:
```bash
${TELEPORT_BINARY_DIR}/teleport node configure \
      --token token \
      joinmethod \
      --ca-pin pin \
      --auth-server host:port \
      "" \
      --output someport
```
That empty argument breaks things.

So, in order to fix it, we are going to change the default value when no
labels are provided.
Instead of an empty string, we'll use an empty array.

Demo (teleport node configure message removed for brev
No label
```bash
$ LABELS_FLAG=(); f=$(mktemp -d)/node.yaml; teleport node configure --auth-server w:1 "${LABELS_FLAG[@]}" --output $f && yq .s
sh_service.labels $f

enabled: "yes"
commands:
  - name: hostname
    command: [hostname]
    period: 1m0s

```

Single label
```bash
$ LABELS_FLAG=(--labels x=y); f=$(mktemp -d)/node.yaml; teleport node configure --auth-server w:1 "${LABELS_FLAG[@]}" --output $f && yq .ssh_service $f

enabled: "yes"
labels:
  x: "y"
commands:
  - name: hostname
    command: [hostname]
    period: 1m0s

```

Multiple labels
```bash
$ LABELS_FLAG=(--labels x=y,dev=prod); f=$(mktemp -d)/node.yaml; teleport node configure --auth-server w:1 "${LABELS_FLAG[@]}" --output $f && yq .ssh_service $f

enabled: "yes"
labels:
  dev: prod
  x: "y"
commands:
  - name: hostname
    command: [hostname]
    period: 1m0s
```

Commit:
Fix some typos in `lib/` (#15556)

Commit:
`SharedDirectoryCreateResponse` update (#15289)

ptgott commented on docs/pages/setup/admin/daemon.mdx:
````diff
@@ -33,89 +31,102 @@ $ readlink /sbin/init
+  ```
 
-Enable the unit so systemd can place it in its dependency tree:
+  Enter the following command to enable the systemd `teleport` daemon. Enabling the daemon using `systemctl` allows systemd to place the daemon in its dependancy tree, ensuring that the teleport service will be automatically restarted on any potential reboot:
````
```suggestion
  Enter the following command to enable the systemd `teleport` daemon. Enabling the daemon using `systemctl` allows systemd to place the daemon in its dependency tree, ensuring that the `teleport` service will be automatically restarted on any potential reboot:
```

ptgott commented on docs/pages/setup/admin/daemon.mdx:
````diff
@@ -33,89 +31,102 @@ $ readlink /sbin/init
+  ```
 
-Start the unit:
+  Next, enter the following command to start the teleport service daemon:
````
```suggestion
  Next, enter the following command to start the `teleport` service daemon:
```

ptgott commented on docs/pages/setup/admin/daemon.mdx:
```diff
@@ -33,89 +31,102 @@ $ readlink /sbin/init
+  </TabItem>
+
+  <TabItem label="Source Installation">
+  In order to more readily support systemd configurations, Teleport is capable of recommending and assisting with the installation of a unit file.
```
Nit: I think we can begin this tab with the next sentence and remove this one without any loss in meaning.

ptgott commented on docs/pages/setup/admin/daemon.mdx:
```diff
@@ -33,89 +31,102 @@ $ readlink /sbin/init
 
-The next time you restart your host, `systemd` will run the `teleport` daemon
-automatically.
+  Enable the `teleport` service daemon using `systemctl` so that systemd can place it in its dependancy tree. This also ensures that the `teleport` service will be automatically restarted on any potential reboot:
```
```suggestion
  Enable the `teleport` service daemon using `systemctl` so that systemd can place it in its dependency tree. This also ensures that the `teleport` service will be automatically restarted on any potential reboot:
```

ptgott commented on docs/pages/setup/admin/daemon.mdx:
```diff
@@ -143,15 +154,44 @@ To upgrade a host to a newer version of Teleport, you must:
+
+| Flag | Description | General Recommended Default |
+| --- | ----------- | ---- |
+| --fd-limit | The maximum number of open file descriptors, definted by `LimitNOFILE` in the unit file. |  **--fd-limit=8192** |
```
```suggestion
| `--fd-limit` | The maximum number of open file descriptors, defined by `LimitNOFILE` in the unit file. |  `--fd-limit=8192` |
```

ptgott commented on docs/pages/setup/admin/daemon.mdx:
```diff
@@ -143,15 +154,44 @@ To upgrade a host to a newer version of Teleport, you must:
+| Flag | Description | General Recommended Default |
+| --- | ----------- | ---- |
+| --fd-limit | The maximum number of open file descriptors, definted by `LimitNOFILE` in the unit file. |  **--fd-limit=8192** |
+| --env-file | The path to the `EnvironmentFile` defined by the system unit file, containing any potential variables to be used in the configuration. | **--envfile=/etc/default/teleport** |
```
```suggestion
| `--env-file` | The path to the `EnvironmentFile` defined by the system unit file, containing any variables to be used in the configuration. | `--envfile=/etc/default/teleport` |
```

ptgott replied:
Also, when would a user want to change this?

ptgott commented on docs/pages/setup/admin/daemon.mdx:
```diff
@@ -143,15 +154,44 @@ To upgrade a host to a newer version of Teleport, you must:
+| --- | ----------- | ---- |
+| --fd-limit | The maximum number of open file descriptors, definted by `LimitNOFILE` in the unit file. |  **--fd-limit=8192** |
+| --env-file | The path to the `EnvironmentFile` defined by the system unit file, containing any potential variables to be used in the configuration. | **--envfile=/etc/default/teleport** |
+| --pid-file | The path to the pid file containing the process identification number. | **--pid-file=/run/teleport.pid** |
```
When would someone want to change this?

rsyracuse replied:
These are largely just to allow for increased customizability and shouldn't deviate from the default configuration often, but do have some potential benefits. The `--fd-limit` and the open file descriptors mentioned relate to memory. Since the open file descriptors have to be suspended in memory this limit sets the bounds for the process. Competing processes or higher expected memory usage by teleport could call for an increase of the --fd-limit. 

 The file paths to the --env-file and --pid-file are good defaults, but on certain distros or other unique configurations someone setting up teleport may find themselves needing to store those files within a different directory either because the path they're trying to use is not a good default, files are stored in a different location, permissions need to be set more finely for services given a certain environment or use case etc.... 
 
 I'm happy to readjust wording around the flags if we wanted or needed, especially if you anticipate the way i'm presenting this information may cause some friction, it's possible there may be context I'm missing.

zmb3 replied:
The `--fd-limit` is the one most likely to need changing. We even document on our scaling page that large clusters should bump it to 65535.

The other two I imagine will work fine with the defaults in almost all cases. We just exposed flags for these because relying on hard-coded paths when there are so many linux distros out there with subtle differences felt like a bad idea.

ptgott commented on docs/pages/setup/admin/daemon.mdx:
```diff
@@ -143,15 +154,44 @@ To upgrade a host to a newer version of Teleport, you must:
+| --- | ----------- | ---- |
+| --fd-limit | The maximum number of open file descriptors, definted by `LimitNOFILE` in the unit file. |  **--fd-limit=8192** |
+| --env-file | The path to the `EnvironmentFile` defined by the system unit file, containing any potential variables to be used in the configuration. | **--envfile=/etc/default/teleport** |
+| --pid-file | The path to the pid file containing the process identification number. | **--pid-file=/run/teleport.pid** |
```
Like the previous rows, I think it would make more sense to put flag examples in code style to make it explicit that these are possible parts of the `teleport install systemd` command.

ptgott commented on docs/pages/setup/admin/daemon.mdx:
```diff
@@ -143,15 +154,44 @@ To upgrade a host to a newer version of Teleport, you must:
+| --env-file | The path to the `EnvironmentFile` defined by the system unit file, containing any potential variables to be used in the configuration. | **--envfile=/etc/default/teleport** |
+| --pid-file | The path to the pid file containing the process identification number. | **--pid-file=/run/teleport.pid** |
+| --teleport-path | The full path to the `teleport` binary. | **--teleport-path=/usr/local/bin/teleport** |
+| -o, --output | The full unit file load path which will define where to create and store the system unit file itself. This flag is used to write the output to the designated file. | **--output=/etc/systemd/system/teleport.service** |
```
```suggestion
| `-o`, `--output` | The full unit file load path, which will define where to create and store the system unit file itself. This flag is used to write the output to the designated file. | `--output=/etc/systemd/system/teleport.service` |
```

ptgott commented on docs/pages/setup/admin/daemon.mdx:
````diff
@@ -143,15 +154,44 @@ To upgrade a host to a newer version of Teleport, you must:
+If using the recommended defaults, you can enter all command line options in a single command as follows:
+
+```code
+teleport install systemd \
````
```suggestion
$ teleport install systemd \
```
So the docs engine renders the command correctly in the `code` snippet

ptgott reviewed (approved):
Mostly tiny suggestions here, plus two questions about the CLI flag table.

The conflicts are due to merging a PR (#15357) that reorganized the docs. They shouldn't be more than updating link paths.

Commit:
Fix `process_irp_set_information` (#15364)

logand22 committed:
Update helm chart snapshots and tests for amazon ECR Public (#15355)

zmb3 committed:
Build Teleport Connect for Linux (#15509)

In order to do so, we add a new make target:

    make teleterm

This (temporarily) assumes that the gravitational/webapps repo is
cloned at the right version as a sibling to the teleport repo.
(We'll be able to get rid of this when we merge webapps into Teleport)

Additionally, update dronegen to include the name of the calling
function that generated the snippet instead of the line number.
This gets rid of lots of superfluous diffs in the generated
.drone.yml file.

Lastly, rewrite the Go program for getting the right webapps version
in bash, because Go is not available at this step of the drone pipeline.

Co-authored-by: Grzegorz Zdunek <grzegorz.zdunek@goteleport.com>

zmb3 committed:
Fix push builds for Teleport Connect (#15779)

PR #15509 updated the tag pipeline but missed the push pipeline

logand22 committed:
Fix small missing change with docker registry (#15782)

logand22 committed:
Authenticated pulls to build artifacts pipeline (#15781)

rsyracuse commented on docs/pages/setup/admin/daemon.mdx:
```diff
@@ -2,18 +2,21 @@
-Teleport process runs in the foreground. On Linux systems in non-containerized
-environments, we recommend running the `teleport` binary as a daemon using
-systemd.
+We have included a default `teleport` daemon configuration in DEB and RPM installed packages, however you must manually start it using `systemctl`. When you install the Teleport binary from source, a systemd daemon is not pre-configured automatically,
```
Removed this, good observations I can agree with

rsyracuse commented on docs/pages/setup/admin/daemon.mdx:
```diff
@@ -33,89 +36,106 @@ $ readlink /sbin/init
-[Unit]
-Description=Teleport SSH Service
-After=network.target
+Systemd is designed to use **unit files**, which are the primary file used to define and set the conditions for
```
Reworded this a bit. The path of the configuration file helps defines permissions and load order when used as intended. The wording in this paragraph was intended as a nod to that, but I'm avoiding going into greater length since I think that explainer is a little outside of the scope of the issue we're trying to resolve in this guide and could instead cause confusion. I worded this mostly per your recommendations because I think confusion is a risk otherwise. Resolving for now, but let me know if you have additional feedback!

Commit:
spelling/formatting

Commit:
spelling/formatting

Commit:
code formatting

Commit:
code formatting

Commit:
address merge conflict

rsyracuse commented:
Resolution of the merge conflict caused an extraneous commit history. Closing this for now to move to a new PR https://github.com/gravitational/teleport/pull/15800
